api/v1alpha1/shared_types.go

```diff
@@ -243,6 +245,8 @@ type KubernetesServiceSpec struct {
 	// may be ignored if the load balancer provider does not support this feature.
 	// This field has been deprecated in Kubernetes, but it is still used for setting the IP Address in some cloud
 	// providers such as GCP.
+	//
+	// +kubebuilder:validation:XValidation:message="loadBalancerIP must be a valid IPv4 address",rule="self.matches(r\"^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$\")"
```

Comment:
Currently you can only use RE, nothing else.

Reply:
It's unfortunate, since lookahead would have easily worked here to avoid the duplication were it supported.